app/services/bulk_update/add_trainees/validate_file.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require "charlock_holmes"
+
+module BulkUpdate
+  module AddTrainees
+    class ValidateFile
+      def initialize(file:, record:)
+        @file = file
+        @record = record
+      end
+
+      def validate!
+        if file
+          file_size_within_range? && file_encoding_is_accepted? && valid_csv?
+        else
+          record.errors.add(:file, :missing)
+          false
+        end
+      end
+
+    private
+
+      attr_reader :file, :record
+
+      def file_encoding_is_accepted?
+        return true if %w[UTF-8 ISO-8859-1].include?(encoding)
+
+        record.errors.add(:file, :encoding_not_accepted, encoding:)
+      end
+
+      def detection
+        @detection ||= begin
+          contents = File.read(file)
+          CharlockHolmes::EncodingDetector.detect(contents)
+        end
+      end
+
+      def encoding
+        @encoding ||= detection&.dig(:encoding)
+      end
+
+      def file_size_within_range?
+        if file.size > 1.megabyte
+          record.errors.add(:file, :too_large)
+          return false
+        elsif file.size.zero? # rubocop:disable Style/ZeroLengthPredicate
```

Comment:
```suggestion
        elsif file.size.empty? 
```

Maybe that since the error is for `empty`